system-variables: add system variable tidb_gc_max_wait_time (#8720)
pingcap/docs: garbage-collection-configuration.md

* [`tidb_gc_life_time`](/system-variables.md#tidb_gc_life_time-new-in-v50)
* [`tidb_gc_concurrency`](/system-variables.md#tidb_gc_concurrency-new-in-v50)
* [`tidb_gc_scan_lock_mode`](/system-variables.md#tidb_gc_scan_lock_mode-new-in-v50)

> * [`tidb_gc_max_wait_time`](system-variables.md#tidb_gc_max_wait_time-new-in-v610)

Author: ```suggestion
* [`tidb_gc_max_wait_time`](/system-variables.md#tidb_gc_max_wait_time-new-in-v610)
```

For information on changes in previous releases, refer to earlier versions of this document using the _TIDB version selector_ in the left hand menu.

## Changes in TiDB 6.1.0

> Before TiDB v6.1.0, the transaction in TiDB does not affect the GC safe point. Since v6.1.0,  TiDB considers the startTS of the transaction when calculating the GC safe point, to resolve the problem that the data to be accessed has been cleared. If the transaction is too long, the safe point will be blocked for a long time, which affects the application peformance.

Author: ```suggestion
Before TiDB v6.1.0, the transaction in TiDB does not affect the GC safe point. Since v6.1.0, TiDB considers the startTS of the transaction when calculating the GC safe point, to resolve the problem that the data to be accessed has been cleared. If the transaction is too long, the safe point will be blocked for a long time, which affects the application performance.
```

> In versions before TiDB 6.1.0, the internal transaction does not affect the GC safe point. Since TiDB 6.1.0, the internal transaction's startTS is considered when calculating the GC safe point, to resolve the access problem of cleaned data. When the internal transaction is too long, the safe point will be blocked for a long time, which affect the performance of application.

Reviewer: ```suggestion
Before TiDB 6.1.0, the transaction in TiDB does not affect the GC safe point. Since v6.1.0,  TiDB considers the startTS of the transaction when calculating the GC safe point, to resolve the problem that the data to be accessed has been cleared. If the transaction is too long, the safe point will be blocked for a long time, which affects the application peformance.
```

> TiDB v6.1.0 introduces the system variable [`tidb_gc_max_wait_time`](/system-variables.md#tidb_gc_max_wait_time-new-in-v610)to control the maximum time that active transactions block the GC safe point. After the value is exceeded, the GC safe point is forwarded forcefully.

Reviewer: ```suggestion
In TiDB v6.1.0, the system variable [`tidb_gc_max_wait_time`](/system-variables.md#tidb_gc_max_wait_time-new-in-v610) is introduced to control the maximum time that active transactions block the GC safe point. After the value is exceeded, the GC safe point is forwarded forcefully.
```
